### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by JDSweet** [symphony-of-empires/symphony-of-empires#124]
For purposes of longevity, and ease of use for new devs who are interested in getting involved in this project (before it blows up too much more), it might be helpful/useful to write out some algorithm documents that explain at varying levels of detail how the engine and different systems work, what problems were encountered, and what the step-by-step solutions (in a very broad sense) were to those problems.

I'm a uni student, and this is something the professors often require (under the explanation that, as I said, it helps others read and understand the code and how all the systems that make the software work mesh together with one another).

**Comment by coderguy57** [comment]
Yeah your right! We are severely lacking in documentation. We will try to get some better documentation up soon :+1:

**Comment by coderguy57** [comment]
We have this: https://symphony-of-empires.github.io/docs
It's not that great, but we work on improving it
